addon/core/say-data-factory.ts

```diff
@@ -72,18 +80,58 @@ export class SayDataFactory extends DataFactory {
     value: string,
     languageOrDataType?: string | SayNamedNode<string>,
   ): LiteralNodeTerm {
-    const literal = this.literal(value, languageOrDataType);
+    // TODO: we need to be careful with using spread operators here:
+    // the `equals` prototype method is not preserved.
     return {
-      ...literal,
+      ...this.literal(value, languageOrDataType),
       termType: 'LiteralNode',
     };
   }
   resourceNode(value: string): ResourceNodeTerm {
+    // TODO: we need to be careful with using spread operators here:
+    // the `equals` prototype method is not preserved.
     return { ...this.namedNode(value), termType: 'ResourceNode' };
   }
   variable(value: string): SayVariable {
     return super.variable(value);
   }
+
+  fromJson(serialized: Omit<SayTerm, 'equals'>): SayTerm {
+    switch (serialized.termType) {
+      case 'BlankNode':
+        return this.blankNode(serialized.value);
+      case 'NamedNode':
+        return this.namedNode(serialized.value);
+      case 'Literal': {
+        const { datatype, language } = <SayLiteral>serialized;
```

Comment:
For consistency, I prefer the `as` syntax for casting
Will also be a useful habit for when we move to `.gts` components

Reply:
Yeah makes sense, it's the first time I've used it (as the original lib also uses it)